src/features/decoration/views.tsx

```diff
@@ -26,23 +26,31 @@ export class IssueMarkerView implements IView {
     render(marker: SIssueMarker, context: RenderingContext): VNode {
         const scale = 16 / 1792;
         const trafo = `scale(${scale}, ${scale})`;
+        const maxSeverity = this.getMaxSeverity(marker);
         const group = <g class-sprotty-issue={true}>
             <g transform={trafo}>
-                <path d={this.getPath(marker.severity)} />
+                <path d={this.getPath(maxSeverity)} />
             </g>
         </g>;
-        setClass(group, 'sprotty-' + marker.severity, true);
+        setClass(group, 'sprotty-' + maxSeverity, true);
         return group;
     }
 
+    protected getMaxSeverity(marker: SIssueMarker): SIssueSeverity {
+        let currentSeverity: SIssueSeverity = 'info';
+        for (const severity of marker.issues.map(s => s.severity)) {
+            if (severity === 'error' || (severity === 'warning' && currentSeverity === 'info'))
+                currentSeverity = severity;
+        }
+        return currentSeverity;
+    }
+
     protected getPath(severity: SIssueSeverity) {
         switch (severity) {
             case 'error':
-                // tslint:disable-next-line:max-line-length
-                return "M768 128q209 0 385.5 103t279.5 279.5 103 385.5-103 385.5-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103zm128 1247v-190q0-14-9-23.5t-22-9.5h-192q-13 0-23 10t-10 23v190q0 13 10 23t23 10h192q13 0 22-9.5t9-23.5zm-2-344l18-621q0-12-10-18-10-8-24-8h-220q-14 0-24 8-10 6-10 18l17 621q0 10 10 17.5t24 7.5h185q14 0 23.5-7.5t10.5-17.5z";
```

Comment:
So now there's a fall-through to 'warning'? Why?

Reply:
Same icon, different color. Initially I used a triangle icon for warnings, but Theia uses circles for both, so I aligned that